### PR TITLE
chore: suite utils (wallet and few others from suite) and types to common package

### LINKED ISSUE #5801: Move wallet utils to shared common package
That could be used between suite and native app.

### PR DESCRIPTION
**Pull request by bouchja1**
**Warning: Diff is currently messed up, because source branch was rebased. I recommend you to wait with check of this PR until #5799 will be merged and this PR rebased.**

Based on https://github.com/trezor/trezor-suite/pull/5799
- moving stuff around and changing imports according to it. 

new packages in **suite-common**:

- `wallet-utils` - moved utils and related tests from suite/src/utils/wallet (except coinmarket, graph utils and accountUtils related to router params)
- `suite-types` - moved types for TrezorDevice and guides and messageSystem.
- `suite-config` - features config
- `suite-constants`
- `suite-utils` - features
- `test-utils` - mocks (npmMocks and setupJest) moved here from suite/src/support/tests to be shared between packages

changes in code:
- @trezor/suite-data:
	- message-system - messageSystem.ts changed to be generated to suite-common/suite-types
	- guide - Category => GuideCategory - Category from guides renamed to GuideCategory because there was a conflicting interface name Category from messageSystem.ts when both are exported in new `suite-types` now

closes https://github.com/trezor/trezor-suite/issues/5801

### CODE REVIEW

File: suite-common/suite-types/src/device.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    KnownDevice,
+    PROTO,
+    UI,
+    UnknownDevice as UnknownDeviceBase,
+    UnreadableDevice as UnreadableDeviceBase,
+} from '@trezor/connect';
+import { DeviceMetadata } from '@suite-common/metadata-types';
+
+// TODO former type was from suite Action['type'][] - check if this is correct
+const blockedActions: string[] = [UI.CLOSE_UI_WINDOW];
```
Comment (bouchja1) [inline suite-common/suite-types/src/device.ts:12]: check this

Comment (Nodonisko) [reply suite-common/suite-types/src/device.ts:13]: I would prefer some type redux-toolkit or if notthing suitable here, at least alias like `type ActionType = string`

File: suite-common/test-utils/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "@suite-common/test-utils",
```
Comment (Nodonisko) [inline suite-common/test-utils/package.json:2]: Please add README to this package, about how to use this package.

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-common/test-utils/package.json (hunk 2 above)
Comment (bouchja1) [reply suite-common/test-utils/package.json:2]: https://github.com/trezor/trezor-suite/pull/5815/commits/1783c364b7bba52fb55ac38fb98fa191a9fdc17b

**Pull request closed without merging**